spec.html

```diff
@@ -440,6 +440,8 @@
       <p>The <em>syntactic grammar</em> for ECMAScript is given in clauses 11, 12, 13, 14, and 15. This grammar has ECMAScript tokens defined by the lexical grammar as its terminal symbols (<emu-xref href="#sec-lexical-and-regexp-grammars"></emu-xref>). It defines a set of productions, starting from two alternative goal symbols |Script| and |Module|, that describe how sequences of tokens form syntactically correct independent components of ECMAScript programs.</p>
       <p>When a stream of code points is to be parsed as an ECMAScript |Script| or |Module|, it is first converted to a stream of input elements by repeated application of the lexical grammar; this stream of input elements is then parsed by a single application of the syntactic grammar. The input stream is syntactically in error if the tokens in the stream of input elements cannot be parsed as a single instance of the goal nonterminal (|Script| or |Module|), with no tokens left over.</p>
       <p>When a parse is successful, it constructs a <em>parse tree</em>, a rooted tree structure in which each node is a <dfn>Parse Node</dfn>. Each Parse Node is an <em>instance</em> of a symbol in the grammar; it represents a span of the source text that can be derived from that symbol. The root node of the parse tree, representing the whole of the source text, is an instance of the parse's goal symbol. When a Parse Node is an instance of a nonterminal, it is also an instance of some production that has that nonterminal as its left-hand side. Moreover, it has zero or more <em>children</em>, one for each symbol on the production's right-hand side: each child is a Parse Node that is an instance of the corresponding symbol.</p>
+      <p>Each time a stream of code points is parsed, it produces Parse Nodes with a fresh identity. Two Parse Nodes are considered <dfn>the same Parse Node</dfn> if they have the same identity, that is, that they resulted from the same invocation the grammar.</p>
```

Comment:
(In "Parse Nodes represent", the plural doesn't make sense, because the rest of the sentence is singular. You could change it to "the Parse Node represents" or just leave it as "it represents".)

"New ... newly" is redundant.

I can imagine an implementer balking at the suggestion they can't reuse the results of a previous parse, so you might want to add a note saying why it's specified that way. Or at least a cross-reference to places in the spec where the identity of Parse Nodes is important.

Have you convinced yourself that no-reuse doesn't have unintended consequences? I.e., that there aren't places in the spec that already implicitly assume 'reuse' of parse nodes?